cronjobs/github_api_util.py

```diff
@@ -0,0 +1,44 @@
+# Python 3 module
+# -*- coding: utf-8 -*-
+import json
+import os
+import re
+import requests
+import util
+
+def GetAllIssuesForUBTueRepository(repository):
+    if 'GITHUB_OAUTH_TOKEN' in os.environ:
+        github_oauth_token = os.environ.get('GITHUB_OAUTH_TOKEN')
+        headers = { 'Authorization' :  'token ' + github_oauth_token }
+    results = []
+    req = requests.get('https://api.github.com/repos/ubtue/' + repository + '/issues?state=all&per_page=100')
+    for result in req.json():
+        results.append(result)
+    while "next" in req.links:
+        next_url =  req.links['next']['url'];
+        req = requests.get(next_url)
+        for result in req.json():
+            results.append(result)
+    return results
+
+
+def UpdateIssueInRepository(repository, issue_number, data):
+    if not 'GITHUB_OAUTH_TOKEN' in os.environ:
+       raise Exception('GITHUB_OAUTH_TOKEN must be set. Export it from the shell')
+    github_oauth_token = os.environ.get('GITHUB_OAUTH_TOKEN')
+    headers = { 'Authorization' :  'token ' + github_oauth_token }
+    url = 'https://api.github.com/repos/ubtue/' + repository + '/issues/' + issue_number
+    req = requests.post(url, json=data, headers=headers)
+    req.raise_for_status()
+    print(req.text)
+
+
+def GetISSNMatcher():
+    return re.compile('([0-9]{4}-[0-9]{3}[0-9X])')
```

Comment:
Hyphens are optional for ISSN's. From the Zotero forum "Still, though, the hyphens in ISSNs and ISBNs should never be treated as anything but presentational polish."

Reply:
In general yes. However, we evaluate ZEDER data or user generated issue subjects the are geared to ZEDER, where the hyphen is always present. Thus, in this context, omitting '?' in the regex is a safeguard against erroneously extracting PPNs or other arbitrary numbers.

Reply:
In that case, why not duplicate the functionality that we have in our C++ function MiscUtil::IsPossibleISSN?